pkg/k8s/statefulsets.go

```diff
@@ -0,0 +1,57 @@
+package k8s
+
+import (
+	"context"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/celestiaorg/torch/pkg/db/redis"
+)
+
+const queueK8SNodes = "k8s"
+
+// WatchStatefulSets watches for changes to the StatefulSets in the specified namespace and updates the metrics accordingly
+func WatchStatefulSets() {
```

Comment:
I highly recommend returning errors and forming them into a return err <msg info why error>